pkg/storegateway/bucket_stores.go

```diff
@@ -137,6 +140,10 @@ func NewBucketStores(cfg tsdb.BlocksStorageConfig, shardingStrategy ShardingStra
 		return nil, errors.Wrap(err, "create index cache")
 	}
 
+	if u.chunksCache, err = tsdb.NewChunksCache(cfg.BucketStore.ChunksCache, logger, reg); err != nil {
```

Comment:
`NewChunksCache()` can return `nil`. This means the upcoming call to `WithChunksCache()` will overwrite the `noopCache` with `nil`. I don't think that's what we want. I would recommend to guarantee that the `chunksCache` is never `nil` to keep it easier to work with. Keep in mind `tsdb.NewIndexCache()` never return nil (unless in case of error) so that's why we've never experienced it with the index cache.

Reply:
i think nil is ok because the code I have in PR 3968 handles it, but i see your point. 

The changes i did in 01b462f2279e22cdafa997f1d3ba87144542face still require setting a noop cache in both `NewBucketStore` and in `tsdb.NewChunksCache` because there are tests that only use `NewBucketStore` and for those we'd still want a non-nil cache.